### ISSUE TITLE
no-useless-not is unaware of `enabled`/`visible` parameter

### ISSUE THREAD

**Issue by karyon** [playwright-community/eslint-plugin-playwright#260]
this reports "Unexpected usage of not.toBeVisible(). Use toBeHidden() instead", but using toBeHidden can't be used with a parameter.
```js
let visible = false;
// reports no-useless-not
await expect(page.locator('#selector')).not.toBeVisible({ visible });
```
same for `toBeEnabled`.

The lint was added in https://github.com/playwright-community/eslint-plugin-playwright/pull/81. Shortly thereafter, playwright 1.26 was released which added these parameters.

**Comment by mskelton** [comment]
@karyon I added support for this in #264 with a pile of test cases and also support for auto-fixing if a boolean literal is passed as the argument:

```ts
await expect(page.locator('#selector')).not.toBeVisible({ visible: false });
// becomes
await expect(page.locator('#selector')).toBeVisible();

await expect(page.locator('#selector')).not.toBeVisible({ visible: true });
// becomes
await expect(page.locator('#selector')).toBeHidden();

await expect(page.locator('#selector')).toBeVisible({ visible: true });
// becomes
await expect(page.locator('#selector')).toBeVisible();
```

If the argument for `visible` or `enabled` isn't a simple boolean literal, we won't report an error given that we can't determine the value of the provided argument.

Unlikely anyone would do that, but if they do, the rule will auto fix it! 😄

**Comment by karyon** [comment]
That was quick, thanks a lot!
